src/gt4py/next/iterator/embedded.py

```diff
@@ -878,60 +888,53 @@ def make_in_iterator(
 builtins.builtin_dispatch.push_key(EMBEDDED)  # makes embedded the default
 
 
-class LocatedFieldImpl(MutableLocatedField):
-    """A Field with named dimensions/axes."""
+@dataclasses.dataclass(frozen=True)
+class NDArrayLocatedFieldWrapper(MutableLocatedField):
+    """A temporary helper until we sorted out all Field conventions between frontend and iterator.embedded."""
+
+    _ndarrayfield: common.Field
 
     @property
     def __gt_dims__(self) -> tuple[common.Dimension, ...]:
-        return self._axes
-
-    def __init__(
-        self,
-        getter: Callable[[FieldIndexOrIndices], Any],
-        axes: tuple[common.Dimension, ...],
-        dtype,
-        *,
-        setter: Callable[[FieldIndexOrIndices, Any], None],
-        array: Callable[[], npt.NDArray],
-        origin: Optional[dict[common.Dimension, int]] = None,
-    ):
-        self.getter = getter
-        self._axes = axes
-        self.setter = setter
-        self.array = array
-        self.dtype = dtype
-        self.origin = origin
-
-    def __getitem__(self, indices: ArrayIndexOrIndices) -> Any:
-        return self.array()[indices]
+        return self._ndarrayfield.__gt_dims__
+
+    def _promote_tags_to_dims_and_sort(
+        self, named_indices: NamedFieldIndices
+    ) -> Mapping[common.Dimension, FieldIndex | SparsePositionEntry]:
+        return {d: named_indices[d.value] for d in self._ndarrayfield.__gt_dims__}
+
+    def _translate_named_indices(self, _named_indices: NamedFieldIndices) -> common.DomainSlice:
+        named_indices = self._promote_tags_to_dims_and_sort(_named_indices)
```

Comment:
I would inline the function here (with the right type hint explicitly declared for named_indices) unless it results can be cached or something else is done in its body